Bluetooth: Controller: Fix jitter in offsets for Extended and Periodic Advertising

cvinayak opened the pull request:
Updated ticker implementation to return remainder value for
a ticker when enumerating active tickers with time
reservations.

This is required to find offsets and to use the remainder
value to correctly calculate auxiliary offsets to the
microsecond resolution.

Use macro to adjust ticks for jitter stored in remainder
value.

Use remainder value in scheduling the periodic auxiliary
PDUs and use the ticker next slot get interface with
remainder value to fill the auxiliary offsets with
microsecond precision in the primary channel PDUs.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Ticker next slot get to return remainder value

Updated ticker implementation to return remainder value for
a ticker when enumerating active tickers with time
reservations.

This is required to find offsets and to use the remainder
value to correctly calculate auxiliary offsets to the
microsecond resolution.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Use remainder in Ext and Periodic PDU offset

Use remainder value in scheduling the periodic auxiliary
PDUs and use the ticker next slot get interface with
remainder value to fill the auxiliary offsets with
microsecond precision in the primary channel PDUs.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Use macro to adjust ticks based on remainder

Use macro to adjust ticks for jitter stored in remainder
value.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

ppryga commented on subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/ticker.h:
```diff
@@ -32,16 +38,30 @@
 	)
 
+/* Macro to remove ticks and return positive remainder value in microseconds */
+#define HAL_TICKER_REMOVE_JITTER(t, r) \
```
Why macro? Wouldn't a function be better option? It does a type checking.

ppryga commented on subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/ticker.h:
```diff
@@ -32,16 +38,30 @@
+	{ \
+		if ((!(r / 1000000UL)) || (r & BIT(31))) { \
+			t--; \
+			r += 30517578UL; \
```
Please change a magic number into named constant. The same value is repeated in multiple places hence it is a great candidate to be named constant.

ppryga commented on subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/ticker.h:
```diff
@@ -22,7 +28,7 @@
 	)
 
-/* Macro returning remainder in nanoseconds */
+/* Macro returning remainder in picoseconds */
```
Was it an error in former description of the macro?

ppryga commented on subsys/bluetooth/controller/ll_sw/lll_adv.h:
```diff
@@ -129,10 +127,11 @@ struct lll_adv_aux {
+	/* Store used by primary channel PDU event to fill the
 	 * auxiliary offset to this auxiliary PDU event.
 	 */
 	uint32_t ticks_offset;
+	uint32_t us_offset;
```
These members names does not say anything about reference point for these offsets. Hence one is forced to get to struct and luckily read the comment above. I suggest to change these names.
```suggestion
	uint32_t prim_adv_ticks_offset;
	uint32_t prim_adv_us_offset;
```

cvinayak committed:
Bluetooth: Controller: Review rework of jitter in aux offset

Review rework of the fix to remove the jitter in aux offset
and sync offset values.

Force select BT_TICKER_REMAINDER_GET and BT_TICKER_LAZY_GET
features when Extended Advertising and Periodic Advertising
is supported.

Rename ticks and microsecond offset value struct members
for primary PDU event offset (to auxiliary PDU event).

Converted HAL_TICKER_REMOVE_JITTER and HAL_TICKER_ADD_JITTER
macro to functions.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak commented on subsys/bluetooth/controller/Kconfig.ll_sw_split:
```diff
@@ -198,7 +198,7 @@ config BT_CTLR_ADV_PDU_LINK
 config BT_CTLR_ADV_PDU_BACK2BACK
 	bool "Back-to-back transmission of extended advertising trains"
-	depends on BT_CTLR_ADV_EXT && BT_BROADCASTER
+	depends on BT_BROADCASTER && BT_CTLR_ADV_EXT
```
just being consistent with the hierarchy of features
